components/gitpod-protocol/src/permission.ts

```diff
@@ -53,6 +54,9 @@ export namespace Permission {
     /** The permission to monitor the (live) state of a Gitpod installation */
     export const MONITOR: PermissionName = "monitor";
 
+    /** The permission to develop on this running Gitpod installation */
+    export const DEVELOPER: PermissionName = "developer";
```

Comment:
What would be the security concerns if we were not introducing another permission here, but allow everyone to use this?

Reply:
> What would be the security concerns if we were not introducing another permission here, but allow everyone to use this?

[XSS](https://de.wikipedia.org/wiki/Cross-Site-Scripting) - every user on gitpod.io can deliver arbitrary web content to anybody on the web pretending it's us.

I'm not particularly fond of that role, but we need to limit it to Gitpod employees/devs somehow. Alternatively we could have a per-install switch. But I really like the usecase of using this on gitpod.io.

Reply:
Kicked of the authorization discussion [here](https://gitpod.slack.com/archives/C01MLB7J24T/p1674028931887149).

Reply:
In that discussion we decided to:
 - have a secondary, installation-level flag defaulting to `false` which disables this feature
 - set it to true for all dev/staging installations
 - temporarily enable it for gitpod.io, until we have sth that replaces it

Reply:
Added a config flag at `experimental.webapp.proxy.frontendDevEnabled` that defaults to `false`.

Reply:
@geropl we're trying to transition away from it and move to "properly" supported config. The `tl;dr` is - I'm happy with adding this parameter because it's part of a config block. When I come to deprecating that whole block, I'll pull that in.

Longer term, we don't want to be adding things to the `experimental` section **EXCEPT** in cases where a config field actually is a truly experimental feature. A good example is #15658, which is something that we know we need to do, but we don't actually have a good, generic solution that covers all cloud providers - we have put a 90 day limit on this parameter being in the config, with the aim of using that period to see how this feature works and how we can make a good, generic solution.

Hope that clarifies

Reply:
Thank you @mrsimonemms , that helped a lot!
I now understood that:
 - we only flatten config _where it makes sense_ (was unsure about this after reading the doc)
 - that example of a "good experimental usecases" also helped to shape the idea
:pray:

Reply:
Great.

The config doc is a living document. If you can express something better than what I've done, please do so. And I'm always happy to have a cup of tea and discuss something with you